src/rgw/rgw_lc.cc

```diff
@@ -1287,22 +1345,24 @@ int RGWLC::process(int index, int max_lock_secs)
 
 void RGWLC::start_processor()
 {
-  worker = new LCWorker(this, cct, this);
-  worker->create("lifecycle_thr");
+  auto maxw = cct->_conf->rgw_lc_max_worker;
+  for (int ix = 0; ix < maxw; ++ix) {
+    auto worker = new LCWorker(this /* dpp */, cct, this);
+    worker->create((string{"lifecycle_thr_"} + to_string(ix)).c_str());
+    workers.push_back(worker);
```

Comment:
consider storing these workers in a vector of unique_ptrs - currently this memory from `new LCWorker` isn't being freed by `stop_processor()`

nit: this would also leak if the `worker->create()` call were to throw an exception. you could make this exception-safe by allocating with `make_unique()`

Reply:
i used unique_ptr in an earlier version, I'll bring it back